### ISSUE TITLE
Requests format for uploading multiple images

### ISSUE THREAD

**Issue by hoangpm-2081** [tiangolo/fastapi#2221]
### Example

Here's my code trying to upload a list of images:

```Python
import requests
import glob
import cv2

path = glob.glob("test_folder/*", recursive=True) # a list of image's path

lst_img = []
for p in path[:3]:
    # img = cv2.imread(p)
    lst_img.append((p, open(p, 'rb'), "image/jpeg"))

data = {"files": lst_img}

url = "http://localhost:6789/" # url api of app

res = requests.post(url=url, data=data)

print(res.status_code)

print(res.text)

```

### Description

I am trying to upload a list of images via url api. Here, i use requests (python package) but maybe my request's format is wrong then i cannot post to url api. I get error code 422:

```bash
"detail":[{"loc":["body","files",0],"msg":"Expected UploadFile, received: <class 'str'>","type":"value_error"}
```

This is my request's format:

```bash
{'files': [('test_folder/image77.jpeg', <_io.BufferedReader name='test_folder/image77.jpeg'>, 'image/jpeg'), ('test_folder/image84.jpeg', <_io.BufferedReader name='test_folder/image84.jpeg'>, 'image/jpeg'), ('test_folder/image82.jpeg', <_io.BufferedReader name='test_folder/image82.jpeg'>, 'image/jpeg')]}
```

I try many ways but always fails. Many thank if u guys help to solve it.

### Environment

* OS: Linux: (Ubuntu 18.04)
* FastAPI Version: 0.61.1
* Requests Version: 2.24.0
* Python Version: 3.7.5

My FastAPI main.py
```
from typing import List

from fastapi import FastAPI, File, UploadFile
from fastapi.responses import StreamingResponse, FileResponse

app = FastAPI()

@app.post("/")
async def main(files: List[UploadFile] = File(...)):
    # file_like = open(video_path, mode="rb")
    # return StreamingResponse(file_like, media_type="video/mp4")
    return {"filenames": [file.filename for file in files]}
```

**Comment by transfluxus** [comment]
did you try adding the files with the files argument?

https://2.python-requests.org/en/master/user/quickstart/#post-a-multipart-encoded-file

**Comment by hoangpm-2081** [comment]
> did you try adding the files with the files argument?
> 
> https://2.python-requests.org/en/master/user/quickstart/#post-a-multipart-encoded-file

Yes, i try many ways by following requests docs, but it's still error

```
lst_img.append(("file", (p, open(p, 'rb'), "image/jpeg")))
```

**Comment by transfluxus** [comment]
can you show your server code?

**Comment by hoangpm-2081** [comment]
> can you show your server code?
> I have something like this:
> 
> ```
> file = {"file": ("avatar.jpg", open("avatar.jpg"), "rb"), "image/jpg")}
> resp = requests.post(urljoin(base_path, "profile_pic"), files=file)
> ```

I added my server code on top of topic, and i see u post 1 file not a list of file then can u see my code again. Thank you

**Comment by transfluxus** [comment]
the post parameter files is a dict, the names above is not the best but you can add multiple files there. 
the keys of the dict need to correspond with the parameters name in the post.
Full example

**server:**

```
import uvicorn
from fastapi import FastAPI, File, UploadFile

app = FastAPI()

@app.post("/files")
async def file_upload(
    avatar: UploadFile = File(...),
	banner: UploadFile = File(...)
):
	print(avatar)
	print(banner)
	open("fd.jpg","wb").write(avatar.file.read())
	return "top"

if __name__ == "__main__":
	uvicorn.run(app)

```

**client:**

```
import requests

files = {"avatar": ("avatar.jpg", open("avatar.jpg", "rb"), "image/jpg"),
         "banner": ("banner.jpg", open("banner.jpg", "rb"), "image/jpg")}
res = requests.post("http://localhost:8000/files", files=files)
print(res.content)
```

Indeed I don't know how you could send a _list_ of images

**Comment by transfluxus** [comment]
ah, I found a way. you can do

```
from starlette.requests import Request


@app.post("/files")
async def file_upload(
    r: Request
):
	print(r)
	images = (await r.form())._dict
```

in request you still need to use a dict, but converting list_index to keys should be no problem. But I guess arbitrary list of files goes over the scope of this issue. just in case somebody wants to know :)

**Comment by hoangpm-2081** [comment]
> the post parameter files is a dict, the names above is not the best but you can add multiple files there.
> the keys of the dict need to correspond with the parameters name in the post.
> Full example
> 
> **server:**
> 
> ```
> import uvicorn
> from fastapi import FastAPI, File, UploadFile
> 
> app = FastAPI()
> 
> @app.post("/files")
> async def file_upload(
>     avatar: UploadFile = File(...),
> 	banner: UploadFile = File(...)
> ):
> 	print(avatar)
> 	print(banner)
> 	open("fd.jpg","wb").write(avatar.file.read())
> 	return "top"
> 
> if __name__ == "__main__":
> 	uvicorn.run(app)
> ```
> 
> **client:**
> 
> ```
> import requests
> 
> files = {"avatar": ("avatar.jpg", open("avatar.jpg", "rb"), "image/jpg"),
>          "banner": ("banner.jpg", open("banner.jpg", "rb"), "image/jpg")}
> res = requests.post("http://localhost:8000/files", files=files)
> print(res.content)
> ```
> 
> Indeed I don't know how you could send a _list_ of images

finally i solved it, thank you for support me

**Comment by MuhammadAbdullah54321** [comment]
How you solved it ??
can you please explain

**Comment by SelanDeemantha** [comment]
Hi @MuhammadAbdullah54321  did you find out a solution for this issue ?
